### DISCUSSION TITLE
Get ordinal suffix for date: i.e. Nov 13th

### DISCUSSION THREAD

**Discussion by mpmdean** [typst/typst#2676] [Q&A] [answered]
If I had a date formatted as follows
```
datetime.today().display("[month repr:long] [day], [year]")
```

which readers as "November 13, 2023" how do I update it to include the  ordinal suffix for date: i.e. November 13th, 2023?

**Comment by frozolotl** [comment, marked as answer]
Sadly, there is no such functionality included yet. I believe date/time formatting will likely need an overhaul or a suitable package in the future.
However, it can still be done in user code:
```typ
#let format-date(date) = {
  let day = date.day()
  let ord = if 10 < day and day < 20 {
    "th"
  } else if calc.rem(day, 10) == 1 {
    "st"
  } else if calc.rem(day, 10) == 2 {
    "nd"
  } else if calc.rem(day, 10) == 3 {
    "rd"
  } else {
    "th"
  }
  [#date.display("[month repr:long]") #day#ord, #date.year()]
}
#format-date(datetime.today())
```

**Reply by extua** [reply]
Just a plug but if you want ordinals in any other context, check out the `nth` package:
https://github.com/typst/packages/tree/main/packages/preview/nth/0.2.0

**Comment by mpmdean** [comment]
Thanks @frozolotl !
